static/js/ducks/sources.js

```diff
@@ -79,7 +79,9 @@ const initialState = {
 // Websocket message handler
 messageHandler.add((actionType, payload, dispatch) => {
   if (actionType === REFRESH_FAVORITE_SOURCES) {
-    dispatch(fetchFavoriteSources());
+    if (window.location.pathname === "/favorites") {
```

Comment:
I think this is fine; may be worth briefly looking at React Router (e.g. https://reactrouter.com/web/api/Hooks/uselocation) to see which is best to use.

Reply:
I think that can only be used inside React components